js/views/modals/Settings/BulkCoinUpdateBtn.js

```diff
@@ -0,0 +1,101 @@
+import $ from 'jquery';
+import baseVw from '../../baseVw';
+import loadTemplate from '../../../utils/loadTemplate';
+import {
+  isBulkCoinUpdating,
+  events as bulkCoinUpdateEvents,
+} from '../../../utils/bulkCoinUpdate';
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    const opts = {
+      ...options,
+      initialState: {
+        isBulkCoinUpdating: isBulkCoinUpdating(),
+        ...options.initialState,
+      },
+    };
+    super(opts);
+
+    this.listenTo(bulkCoinUpdateEvents, 'bulkCoinUpdateDone bulkCoinUpdateFailed',
+      () => this.setState({ isBulkCoinUpdating: false }));
+
+    this.boundOnDocClick = this.onDocumentClick.bind(this);
+    $(document).on('click', this.boundOnDocClick);
+  }
+
+  events() {
+    return {
+      'click .js-applyToCurrent': 'clickApplyToCurrent',
+      'click .js-applyToCurrentCancel': 'clickApplyToCurrentCancel',
+      'click .js-applyToCurrentConfirm': 'clickApplyToCurrentConfirm',
+    };
+  }
+
+  startProcessingTimer() {
+    if (!this.processingTimer) {
+      this.processingTimer = setTimeout(() => {
+        this.processingTimer = null;
+        this.setState({ isBulkCoinUpdating: isBulkCoinUpdating() });
```

Comment:
Can this just be:

```
if (!isBulkCoinUpdating()) this.setState({ isBulkCoinUpdating: false });
```

Otherwise the way you're doing it is you're ensuring the call takes at least 500ms, then if it's still going you'll set the `isBulkCoinUpdating` state to true which will result in another timeout of at least 500ms. At this point we don't need to slap another 500ms onto the call... it's already gone at least that amount, so it's best to let it end when it ends.